WORKSPACE

```diff
@@ -0,0 +1,25 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+#################################
+#### Bazelized external libs ####
+local_repository( name = "xbyak" , path = "src/camlsnark_c/libsnark-caml/depends/xbyak" )
+# http_archive(
+#     name = "xbyak",
+#     urls = ["https://github.com/obazl/xbyak/archive/bzl-1.0.tar.gz"],
+#     strip_prefix = "xbyak-bzl-1.0",
+#     # sha256 = 
+# )
+
+##########################################
+######## Non-bazel external repos ########
+
+## boost needed by: @xbyak//sample:calc
+git_repository(
```

Comment:
Am I right in thinking that builds will need an internet connection for this to work properly? It might be better to add this as a submodule and use a `path` directive instead.

Reply:
On the first point, see [Offline Builds](https://docs.bazel.build/versions/master/external.html#offline-builds).  On the second, see following comment.

Reply:
Also, note that we are mimicking a submodule by putting xbyak's WORKSPACE file in `src/camlsnark_c/libsnark-caml/depends/xbyak`.  Basically that gives us an embedded Bazel repo, since a WORKSPACE file determines a repo.  What that means is that you can only build the xbyak targets by using a repo-qualified label, e.g. @xbyak//test.  If you try `bazel test src/camlsnark_c/libsnark-caml/depends/xbyak/test`, Bazel will complain that you've crossed repo boundaries.  I'll add something to this effect to the documentation.